modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/DataSet.js

```diff
@@ -337,21 +337,34 @@ const DataSet = ({
 		});
 	};
 
-	useEffect(() => {
-		setDataLoading(true);
+	useEffect(
+		() => {
+			setDataLoading(true);
 
-		requestData().then(({data, ok, status: statusCode}) => {
-			if (isMounted()) {
-				if (!ok) {
-					handleApiError({data, statusCode});
-				}
-				else {
-					updateDataSetItems(data);
+			if (apiURL) {
+				requestData().then(({data, ok, status: statusCode}) => {
+					if (isMounted()) {
+						if (!ok) {
+							handleApiError({data, statusCode});
+						}
+						else {
+							updateDataSetItems(data);
+						}
+						setDataLoading(false);
+					}
+				});
+			}
+			else {
+				if (isMounted()) {
+					updateDataSetItems(items);
+
+					setDataLoading(false);
 				}
-				setDataLoading(false);
 			}
-		});
-	}, [isMounted, requestData, setDataLoading]);
+		},
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
`exhaustive-deps` is a pretty important lint rule, [recommended by React team to prevent bugs](https://reactjs.org/docs/hooks-faq.html#what-exactly-do-the-lint-rules-enforce). Here is a [summary of discussion on the subject](https://github.com/facebook/react/issues/14920#issuecomment-471070149), with a lot more details. In short, any variable not listed can end up stuck in state closure. Even if this is unlikely to happen with current code, this can easily result in a hard-to-find bug in the future. It is a good practice to apply this rule, unless there is a very good reason to override it.